File: backend/src/main/java/pe/edu/untels/biblioteca/service/AutorService.java
package pe.edu.untels.biblioteca.service;

import pe.edu.untels.biblioteca.dto.request.AutorRequest;
import pe.edu.untels.biblioteca.dto.response.AutorResponse;

import java.util.List;

public interface AutorService {

    List<AutorResponse> listar();

    AutorResponse obtenerPorId(Long id);

    AutorResponse crear(AutorRequest request);

    AutorResponse actualizar(Long id, AutorRequest request);

    void eliminar(Long id);
}

File: frontend/src/app/core/services/autor.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Autor, AutorRequest } from '../models/autor.model';

@Injectable({ providedIn: 'root' })
export class AutorService {
  private readonly baseUrl = `${environment.apiUrl}/autores`;

  constructor(private readonly http: HttpClient) {}

  listar(): Observable<Autor[]> {
    return this.http.get<Autor[]>(this.baseUrl);
  }

  obtenerPorId(id: number): Observable<Autor> {
    return this.http.get<Autor>(`${this.baseUrl}/${id}`);
  }

  crear(request: AutorRequest): Observable<Autor> {
    return this.http.post<Autor>(this.baseUrl, request);
  }

  actualizar(id: number, request: AutorRequest): Observable<Autor> {
    return this.http.put<Autor>(`${this.baseUrl}/${id}`, request);
  }

  eliminar(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }
}
